logbook-ktor-server/src/main/kotlin/org/zalando/logbook/server/LogbookServer.kt

```diff
@@ -32,25 +29,37 @@ class LogbookServer(
     companion object : BaseApplicationPlugin<Application, Config, LogbookServer> {
         private val responseProcessingStageKey: AttributeKey<Logbook.ResponseProcessingStage> =
             AttributeKey("Logbook.ResponseProcessingStage")
+        private val proceedWithKey: AttributeKey<ByteReadChannel> =
+            AttributeKey("Logbook.proceedWith")
+
         override val key: AttributeKey<LogbookServer> = AttributeKey("LogbookServer")
+
         override fun install(pipeline: Application, configure: Config.() -> Unit): LogbookServer {
             val config = Config().apply(configure)
             val plugin = LogbookServer(config.logbook)
 
-            pipeline.receivePipeline.intercept(ApplicationReceivePipeline.Before) {
+            // This interceptor is executed before ApplicationReceivePipeline,
+            // and regardless of whether the request has body or not
+            pipeline.intercept(ApplicationCallPipeline.Call) {
                 val request = ServerRequest(call.request)
                 val requestWritingStage = plugin.logbook.process(request)
-                val proceedWith = when {
-                    request.shouldBuffer() && !call.request.receiveChannel().isClosedForRead -> {
```

Comment:
Any reason you removed the `request.shouldBuffer()` check? As I think I understand, the `Logbook.process()` decides wether or not the request should be buffered by changing the state of the `HttpRequest` object.

Reply:
It wasn't used anywhere except in this piece of code. Once I re-ordered the code in the way it is, I didn't see ay use for it and all the tests pass.

I'm not sure what it does. `Logbook.process()` is called before, so it cannot depend on it.

Can you provide a unit-test that fails without it?

Reply:
I don't have any unit-test, I was just wondering, because I thought call to `Logbook.process` configures the request `State` depending on the configuration (including the buffer state).
But I might just be wrong, I just guessed it by reading the code.

Reply:
Alright, I'm not the Logbook expert there so I'll let you and the others decide about it ^^